### ISSUE TITLE
Feature Request: group options in the :set screen using <details> and <summary>

### ISSUE THREAD

**Issue by leo848** [qutebrowser/qutebrowser#7192]

**Comment by leo848** [comment]
Of course. When I go to qute://settings (or type in the :set command without arguments), all options available are displayed at once. 

It would be much nicer if options could be displayed in groups, the simplest option would be to just group them by the first name (i.e. 'aliases', 'auto-save', 'backend', 'bindings', 'colors', etc.).

These should be collapsible in groups, one example of implementing this I know would be using the tags <details> and <summary>, but of course more could be done using javascript or slightly more complex HTML.

Right now the screen is very overwhelming in my opinion, and this would be a possible way to fix it.

**Comment by The-Compiler** [comment]
Note that the settings page was already redesigned by @twigleingrid in #6818 and #6999. IMHO that already helped a lot to make it more digestable.

I think it's good to have a page where all settings are visible and searchable, so I'm not sure if collapsing them into sections would make things better really from an UI perspective. Maybe something like a filter would be more appropriate?
